Refactor: Separate Copilot Adapter Logic and Enhance stats UI

bassamsdata opened the pull request:
## Description

- Refactored Copilot adapter by moving token management, model fetching, and stats logic into a new `copilot_helper` module (`lua/codecompanion/helpers/copilot_helper.lua`).
- Functions moved: `get_models`, `get_copilot_stats`, `show_copilot_stats`, and `reset_cache` are now in the helper module.
- The adapter now delegates these responsibilities to the helper, improving separation of concerns and maintainability.
- Separated Copilot stats and helper-related tests into `tests/helpers/test_copilot_helper.lua` (from the original adapter test file).


Stats UI Enhancements:
- Added visual progress bars (9 lines of code) to the Copilot stats window for a clearer, more intuitive display of usage.
- The stats UI now also displays whether overage usage is permitted for premium interactions.



Please let me know if there are any other functions you’d like moved to the helper, if you want the progress bar removed, or if any other adjustments are needed! And if you don’t like this PR at all, that’s completely fine :)


## Related Issue(s)

[here in issue #1717](https://github.com/olimorris/codecompanion.nvim/discussions/1717#discussioncomment-13589180)

## Screenshots

Stats UI

![Screenshot 2025-06-28 at 12 49 51 AM](https://github.com/user-attachments/assets/50764bab-8ba6-4d02-a340-20be9c94a8a8)


Thank you

## Checklist

- [x] I've read the [contributing](https://github.com/olimorris/codecompanion.nvim/blob/main/CONTRIBUTING.md) guidelines and have adhered to them in this PR
- [ ] I've updated `CodeCompanion.has` in the [init.lua](https://github.com/olimorris/codecompanion.nvim/blob/main/lua/codecompanion/init.lua#L239) file for my new feature
- [x] I've added [test](https://github.com/olimorris/codecompanion.nvim/blob/main/CONTRIBUTING.md#testing) coverage for this fix/feature
- [ ] I've updated the README and/or relevant docs pages
- [x] I've run `make all` to ensure docs are generated, tests pass and my formatting is applied

bassamsdata commented on lua/codecompanion/helpers/copilot_helper.lua:
```diff
@@ -0,0 +1,235 @@
+---Fcuntion to reset the Copilot cache
+---@return nil
+function M.reset_cache()
+  _cached_adapter = nil
```
Yeah, I don’t think this is correct. It may seem right, but it’s not.

olimorris replied:
Classic Copilot. The confidence of a senior developer with 20+ years experience. The capability of a developer with 20 months experience.

bassamsdata commented:
It really hurts to see reviews about spelling mistakes. Is there any local version of this to help catch them beforehand? :(

olimorris commented:
@bassamsdata haha welcome to my club 😆.

Yeah I use [Typos](https://github.com/crate-ci/typos?tab=readme-ov-file#install). You can run it locally

olimorris commented:
Thanks for this btw. Looks soooooo nice. You skills with the Neovim UI are really, really impressive.

I was going to propose keeping the Copilot helpers in the adapter folder:

```
adapters
L copilot
    L init.lua
    L helpers.lua 
L anthropic.lua
L openai.lua
```

It feels more logical to keep everything in there and avoids the helpers folder becoming the default storage place for anything that becomes too bulky.

I don't think that will mess up any adapter config.

olimorris commented:
> Sure! I moved them into the new directory as you suggested, and it worked immediately with no issues. The tests have been moved similarly to maintain the same structure.

Amazing. Thank you so much.

> This branch has conflicts that must be resolved
> Use the command line to resolve conflicts before continuing.

These are the words that scare me the most 😆. Truly, truly terrifying. My git skills are so bad. 

> Would you like me to update this branch with main and resolve the conflicts with rebase?

This is where my git knowledge really falls down...this will presumably be okay as we're on a branch in this PR?!

You can also undo any changes to `codecompanion.txt` as well. That nearly always causes merge conflicts due to the damn datetime stamp at the top.

bassamsdata commented:
> These are the words that scare me the most 😆. Truly, truly terrifying. My git skills are so bad.

Mine aren’t great either! :)

> this will presumably be okay as we're on a branch in this PR?!

Yes, I rebased and resolved conflicts since I moved files before you added the "agent" one. Force-pushing is safe because this PR is on my fork’s branch.

> You can also undo any changes to codecompanion.txt as well.

Sure, I did exactly that. Thank you!

Now it’s working with the new structure.
